lib/gltfPipeline.js

```diff
@@ -0,0 +1,108 @@
+'use strict';
+var fs = require('fs');
+var addDefaults = require('./addDefaults');
+var removeUnusedImages = require('./removeUnusedImages');
+var removeUnusedSamplers = require('./removeUnusedSamplers');
+var removeUnusedShaders = require('./removeUnusedShaders');
+var removeUnusedTechniques = require('./removeUnusedTechniques');
+var removeUnusedPrograms = require('./removeUnusedPrograms');
+var removeUnusedBuffers = require('./removeUnusedBuffers');
+var removeUnusedBufferViews = require('./removeUnusedBufferViews');
+var removeUnusedMaterials = require('./removeUnusedMaterials');
+var removeUnusedSkins = require('./removeUnusedSkins');
+var removeUnusedCameras = require('./removeUnusedCameras');
+var removeUnusedTextures = require('./removeUnusedTextures');
+var removeUnusedMeshes = require('./removeUnusedMeshes');
+var removeUnusedNodes = require('./removeUnusedNodes');
+var removeUnusedAccessors = require('./removeUnusedAccessors');
+var removeUnused = require('./removeUnused');
+var parseBinaryGltf = require('./parseBinaryGltf');
+var addPipelineExtras = require('./addPipelineExtras');
+var convertDagToTree = require('./convertDagToTree');
+var combineMeshes = require('./combineMeshes');
+var combinePrimitives = require('./combinePrimitives');
+var removeUnusedVertices = require('./removeUnusedVertices');
+var OptimizationStatistics = require('./OptimizationStatistics');
+var writeGltf = require('./writeGltf');
+var writeBinaryGltf = require('./writeBinaryGltf');
+var readGltf = require('./readGltf');
+var Cesium = require('cesium');
+var defined = Cesium.defined;
+var defaultValue = Cesium.defaultValue;
+
+module.exports = {
+    processJSON : processJSON,
+    processFile : processFile,
+    processJSONToDisk : processJSONToDisk,
+    processFileToDisk : processFileToDisk
+};
```

Comment:
In order to handle the case where the user has a gltf json in memory but not with the extras, `processJSONToDisk` and `processJSON` should take in `gltf` instead of `gltfWithExtras`. They can add the extras to the gltf then call a function named `processJSONWithExtras` which actually does the pipeline steps (like what `processJSON` does currently). The new `processJson` could also remove pipeline extras before sending back the gltf.